### PR TITLE
Fix problem switching from Uint16 to Uint32 indices for outlining.

### PR DESCRIPTION
**Pull request by kring**
Adding edge outlines usually requires adding new vertices, and when we're using a UNSIGNED_SHORT index buffer and the new vertices push us over 65536 vertices, we need to upgrade to an UNSIGNED_INT index buffer. I had previously added this capability, and wrote tests for it, but didn't have any real-world data to test it with at the time. Unfortunately, that code doesn't actually work, and will corrupt rendering of the model when it's triggered. It was all part of my plan to serve as a good illustration for everyone of the dangers of only writing tests for some code and not actually trying it with real data. 😆 

The problem is that `Model` also squirrels away the bufferView's `componentType` in `ModelLoadResources.indexBuffersToCreate` and then uses _that_ to create the buffer instead of the accessor's `componentType` in the glTF. So I'm also helping to illustrate the dangers of denormalization. 👍

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @kring!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by mramato** [comment]
What the plan for this PR? @lilleyse who would be the best to review this?

@emackey you're the king of glTF edge cases, any thoughts?

**Comment by emackey** [comment]
> and the new vertices push us over 65536 vertices

Sorry, weird edge case here, but the max is 65535 vertices, index numbers 0 through 65534.  The last 16-bit value, 65535 (or -1), is reserved as a "primitive restart" value in some APIs and in glTF.  So when you reach that value, you must switch to 32-bit indexing.

**Comment by kring** [comment]
Oh interesting, I didn't realize that, @emackey! I _think_ it doesn't actually matter here because WebGL doesn't do primitive restart, and this is Cesium's in-memory representation, not a proper glTF. But it can't hurt to change to avoid possible subtle bugs later.

**Comment by kring** [comment]
Done!

**Comment by mramato** [comment]
Like I said, king of glTF edge cases :crown: :smile:

File: Source/Scene/ModelOutlineLoader.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ import TextureMinificationFilter from "../Renderer/TextureMinificationFilter.js"
 import TextureWrap from "../Renderer/TextureWrap.js";
 import ForEach from "../ThirdParty/GltfPipeline/ForEach.js";
 
+// glTF does not allow an index value of 65535 because this is the primitive
+// restart value in some APIs.
+const MAX_GLTF_UINT16_INDEX = 65534;
```
Comment (emackey) [inline Source/Scene/ModelOutlineLoader.js:13]: Travis CI says:

> ^ Unexpected token: keyword «const»

Comment (kring) [reply Source/Scene/ModelOutlineLoader.js]: 😱 whoops, fixed!

**Pull request merged**